test/unit/spec/common/jobManager-Spec.js

```diff
@@ -1008,36 +1082,47 @@ define([
                             });
                         }).then(() => {
                             expect(this.bus.emit).toHaveBeenCalled();
-                            // TODO: improve this test
-                            expect(this.jobManagerInstance.updateModel).toHaveBeenCalledTimes(
-                                Object.values(message).length
-                            );
                             const storedJobs =
                                 this.jobManagerInstance.model.getItem('exec.jobs.byId');
-                            const updateRequests = [];
+                            const updateRequests = [],
+                                updatedJobs = [];
                             Object.values(message).forEach((retryBlob) => {
                                 expect(storedJobs[retryBlob.job_id]).toEqual(
                                     retryBlob.job.jobState
                                 );
                                 expect(storedJobs[retryBlob.retry_id]).toEqual(
                                     retryBlob.retry.jobState
                                 );
+                                ['job', 'retry'].forEach((jobType) => {
+                                    updatedJobs.push(retryBlob[jobType].jobState);
+                                });
                                 if (channelType === jcm.CHANNEL.JOB) {
+                                    const channelString = this.jobManagerInstance._encodeChannel(
+                                        jcm.CHANNEL.JOB,
+                                        retryBlob.retry_id
+                                    );
                                     expect(
-                                        this.jobManagerInstance.listeners[retryBlob.retry_id][
+                                        this.jobManagerInstance.listeners[channelString][
                                             jcm.MESSAGE_TYPE.STATUS
                                         ]
                                     ).toBeDefined();
                                 }
-                                // expect start_update requests for the retried job
-                                updateRequests.push([
-                                    jcm.MESSAGE_TYPE.START_UPDATE,
-                                    { [jcm.PARAM.JOB_ID]: retryBlob.retry_id },
-                                ]);
+                                // expect a status request for the retried job
+                                updateRequests.push(retryBlob.retry_id);
                             });
-                            expect(this.jobManagerInstance.bus.emit.calls.allArgs()).toEqual(
-                                jasmine.arrayWithExactContents(updateRequests)
-                            );
+                            expect(this.jobManagerInstance.bus.emit.calls.allArgs()).toEqual([
```

Comment:
The result of the alterations to the retry job handler: just one job status request sent with a list of job IDs, and one call made to update the model.